model/note/step.go

```diff
@@ -181,10 +195,35 @@ func saveSteps(inst *instance.Instance, steps []Step) error {
 			return err
 		}
 	}
-	// TODO purge the old steps
 	return nil
 }
 
+func purgeOldSteps(inst *instance.Instance, fileID string) {
+	t := time.Now().Add(-cleanStepsAfter)
+	id := stepID(fileID, timeToVersion(t))
+	var steps []Step
+	req := couchdb.AllDocsRequest{
+		Limit:  1000,
+		EndKey: id,
+	}
+	if err := couchdb.GetAllDocs(inst, consts.NotesSteps, &req, &steps); err != nil {
+		inst.Logger().WithField("nspace", "notes").
+			Warnf("Cannot purge old steps for file %s: %s", fileID, err)
+		return
+	}
+	if len(steps) == 0 {
+		return
+	}
+	docs := make([]couchdb.Doc, len(steps))
+	for i := range steps {
+		docs[i] = &steps[i]
+	}
+	if err := couchdb.BulkDeleteDocs(inst, consts.NotesSteps, docs); err != nil {
```

Comment:
`couchdb.Doc` isn't an interface that `Step` implements? I would have thought we could use `steps` directly in stead of `docs` here.

Reply:
> couchdb.Doc isn't an interface that Step implements?

Yes, it is.

> I would have thought we could use steps directly in stead of docs here.

No, we can't use a `[]Step` as a `[]couchdb.Doc` (even if `Step` implements the `couchdb.Doc` interface).

Reply:
OK. Like you said, it's been a while since last I've done go :)